cloudinit/sources/DataSourceOpenStack.py

```diff
@@ -68,6 +68,12 @@ def __str__(self):
         mstr = "%s [%s,ver=%s]" % (root, self.dsmode, self.version)
         return mstr
 
+    def get_admin_pass(self):
+        if self.metadata and "admin_pass" in self.metadata:
+            password = self.metadata["admin_pass"]
+            expire = False
+        return (password, expire)
```

Comment:
`password` and `expire` are undefined variables if the conditional evaluates to false